react/features/authentication/components/web/LoginDialog.js

```diff
@@ -0,0 +1,310 @@
+            _conference: conference,
+            _configHosts: configHosts,
+            roomName,
+            onSuccess,
+            dispatch
+        } = this.props;
+        const { password, username } = this.state;
+        const jid = toJid(username, configHosts);
+
+        if (conference) {
+            dispatch(authenticateAndUpgradeRole(jid, password, conference));
+        } else {
+            connect(jid, password, roomName)
+                .then(connection => {
+                    if (onSuccess) {
+                        onSuccess(connection);
+                        APP.store.dispatch(showNotification({
+                            descriptionArguments: { room: roomName },
+                            descriptionKey: 'notify.canJoinConference',
+                            titleKey: 'connection.AUTHSUCC'
+                        }));
+                    }
+                });
+        }
+    }
+
+    _onChange: Object => void;
+
+    /**
+     * Callback for the onChange event of the field.
+     *
+     * @param {Object} evt - The static event.
+     * @returns {void}
+     */
+    _onChange(evt: Object) {
+        this.setState({
+            [evt.target.name]: evt.target.value
+        });
+    }
+
+    /**
+     * Renders an optional message, if applicable.
+     *
+     * @returns {ReactElement}
+     * @private
+     */
+    renderMessage() {
+        const {
+            _configHosts: configHosts,
+            _connecting: connecting,
+            _error: error,
+            _progress: progress,
+            t
+        } = this.props;
+
+        const { username, password } = this.state;
+
+        let messageKey;
+
+        const messageOptions = {};
+
+        if (progress && progress >= 0.5) {
+            messageKey = t('connection.FETCH_SESSION_ID');
+        } else if (error) {
+
+            const { name } = error;
+
+            if (name === JitsiConnectionErrors.PASSWORD_REQUIRED) {
+                const { credentials } = error;
+
+                if (credentials
+                    && credentials.jid === toJid(username, configHosts)
+                    && credentials.password === password) {
+                    messageKey = t('dialog.incorrectPassword');
+                }
+            } else if (name) {
+                messageKey = t('dialog.connectErrorWithMsg');
+                messageOptions.msg = `${name} ${error.message}`;
+            }
+        } else if (connecting) {
+
+            messageKey = t('connection.CONNECTING');
+        }
+
+        if (messageKey) {
+            const message = translateToHTML(t, messageKey, messageOptions);
+
+            return (
+                <span>
+                    { message }
+                </span>
+            );
+        }
+
+        return null;
+    }
+
+    /**
+     * Implements {@Component#render}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const placeholder = config.hosts.authdomain
+            ? 'user identity'
```

Comment:
Let's go with a single placeholder "User identifier", but put it in the translations please.

Reply:
Removed condition and used a single value for the placeholder. Also, updated main.json.